### PR TITLE
refactor: share post bar to include link to write freeform

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- There used to be a single element for displaying the button to choose from reading history, but after getting back to this, it felt difficult to make some changes since the orders are a bit confusing while considering the responsiveness.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

Previews:

Screenshot 2023-05-29 at 6 39 27 PM

Screenshot 2023-05-29 at 6 38 37 PM

Screenshot 2023-05-29 at 6 38 58 PM

Screenshot 2023-05-29 at 6 39 15 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1336 #done

### CODE REVIEW

File: packages/shared/src/components/squads/SharePostBar.tsx [hunk 1]
```diff
@@ -59,48 +62,59 @@ function SharePostBar({
...
-          buttonSize={sidebarRendered ? ButtonSize.Small : ButtonSize.Medium}
+      <span className="flex relative flex-row items-center w-full">
+        <ProfilePicture
+          className="m-3"
+          user={user}
+          size="large"
+          nativeLazyLoading
+        />
+        <input
+          ref={urlRef}
```
Comment (capJavert) [inline packages/shared/src/components/squads/SharePostBar.tsx:78]: Why do we need `urlRef` now vs before?

File: packages/shared/src/components/squads/SquadPageHeader.tsx [hunk 2]
```diff
@@ -95,16 +98,28 @@ export function SquadPageHeader({
...
-          'absolute bottom-0 w-full translate-y-1/2 px-6 laptopL:px-0 bg-theme-bg-primary',
+          'relative tablet:absolute flex flex-col tablet:flex-row justify-center items-center pt-8 tablet:p-0 bottom-0 w-full tablet:translate-y-1/2 laptopL:px-0 bg-theme-bg-primary laptop:max-w-[41.5rem]',
           shouldShowHighlightPulse && 'highlight-pulse',
-          MAX_WIDTH,
         )}
       >
+        <Divider />
         <SharePostBar
-          className="w-full"
+          className="w-full max-w-[30.25rem]"
```
Comment (capJavert) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:107]: Can we standardize this size maybe to some rounded value? Non blocking.

**Review by capJavert — APPROVED** [review]
Some non blocking comments that caught my eye, looking good 🤘

**Review by capJavert — CHANGES_REQUESTED** [review]
Oh, caught one edge case when posting is limited to moderators, "New post" button still shows. Should be hidden or we need design input on this case?

image

**Comment by sshanzel** [comment]
> Oh, caught one edge case when posting is limited to moderators, "New post" button still shows. Should be hidden or we need design input on this case?

Oh, nice catch. Let me fix that up quickly 🚀

File: packages/shared/src/components/squads/SharePostBar.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SharePostBar.tsx:78]: Ah, it seems to be not used anymore, I was working on something have probably decided not to purse, will remove this one 🚀

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadPageHeader.tsx:107]: I think this should be acceptable as the equivalent would still be divisible by 4. Though we might want to store these values somewhere so we can standardize the usage. WDYT?z

**Comment by sshanzel** [comment]
Should look like below now 🚢 

Screenshot 2023-05-30 at 11 09 01 AM

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/squads/SquadPageHeader.tsx]: I was thinking of introducing more sizes in our design system like `w-121` class would be equivalent, but not sure if can be standardized inside design.

**Review by capJavert — APPROVED** [review]
👍 nice, it does look little cramped to me, like the bar is missing some right padding for text to look good inside of it, won't block it but if you find it off as well you can check with design. 

image

**Comment by sshanzel** [comment]
> 👍 nice, it does look little cramped to me, like the bar is missing some right padding for text to look good inside of it, won't block it but if you find it off as well you can check with design.
> 
> image

Oh, cool. Thanks for the review. I will verify first before merging.

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadPageHeader.tsx]: Ah, right. That's actually better, I see these values also being used more often now. Let's probably discuss this in the web team meeting and coordinate with the design team what should we label these sizes.

**Comment by sshanzel** [comment]
Should be back to normal now 🥳 

Screenshot 2023-05-30 at 3 59 30 PM

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/squads/SquadPageHeader.tsx]: 🤝 agreed

**Comment by capJavert** [comment]
@sshanzel gg wp 🚀

**Pull request merged**